### ISSUE TITLE
About gradient accumulation in your douban_final.py

### ISSUE THREAD

**Issue by Xie-Minghui** [hanjanghoon/BERT_FP#2]
![image](https://user-images.githubusercontent.com/35483387/128991228-17c3f7cf-1631-4763-81fe-c2f8103b6146.png)
in your code, if args.gradient_accumulation_steps > 1, loss.backward() will not be excuted. But in every step, loss.backward() should be excuted.
The normal gradient accumulation process is as follows:
![image](https://user-images.githubusercontent.com/35483387/128991738-2b16fa24-f3f1-4c16-9e79-a2017e487c89.png)

I don't know if I was wrong.

**Comment by hanjanghoon** [comment]
Sorry for the late reply.
As you said, the gradient accumulation code is not implemented properly.
At first I tried to use it, but I got a new gpu card, so I didn't implement it. 
 
thank you for your opinion. i will modify code soon.
 
--

JangHoon Han
AI Research Engineer   |   Professional
Language Lab   |   LG AI Research, Seoul, Korea
Mobile : (+82)10-8591-1081
lgresearch.ai
 
-----Original Message-----
From: ***@***.***>
To: ***@***.***>;
Cc: ***@***.***>;
Sent: 2021-08-11 (수) 16:58:32 (GMT+09:00)
Subject: [hanjanghoon/BERT_FP] About gradient accumulation in your douban_final.py (#2)
 

in your code, if args.gradient_accumulation_steps > 1, loss.backward() will not be excuted. But in every step, loss.backward() should be excuted.
The normal gradient accumulation process is as follows:
I don't know if I was wrong.
—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by hanjanghoon** [comment]
Sorry for the late reply.
As you said, the gradient accumulation code is not implemented properly.
At first I tried to use it, but I got a new gpu card, so I didn't implement it.

thank you for your opinion. i will modify code soon.

**Comment by hanjanghoon** [comment]
model.zero_grad()                                   # Reset gradients tensors
for i, (inputs, labels) in enumerate(training_set):
    predictions = model(inputs)                     # Forward pass
    loss = loss_function(predictions, labels)       # Compute loss function
    loss = loss / accumulation_steps                # Normalize our loss (if averaged)
    loss.backward()                                 # Backward pass
    if (i+1) % accumulation_steps == 0:             # Wait for several backward steps
        optimizer.step()                            # Now we can do an optimizer step
        model.zero_grad()                           # Reset gradients tensors
        if (i+1) % evaluation_steps == 0:           # Evaluate the model when we...
            evaluate_model()
